form-hoc/src/index.js

```diff
@@ -67,41 +77,41 @@ export default class formHoc extends PureComponent {
   }
 
   handleOnBlur(e) {
-    const {inputProps} = this.props
-    const {checkValidationOnBlur, onBlur} = {inputProps}
+    const { inputProps } = this.props
+    const { checkValidationOnBlur, onBlur } = { inputProps }
     this.checkValidations(this.state.value, checkValidationOnBlur)
     if (typeof onBlur === 'function') {
       this.props.onBlur(e)
     }
   }
 
   handleOnChange(e) {
-    const {inputProps} = this.props
-    const {onChange, getValue, checkValidationOnError} = inputProps
+    const { inputProps } = this.props
+    const { onAfterChange, getValue, checkValidationOnChange } = inputProps
     this.setState({
       value: e.target.value
     })
-    this.checkValidations(e.target.value, checkValidationOnError)
+    this.checkValidations(e.target.value, checkValidationOnChange)
     /**
     * you guyz don't need to check for the value
     * call only if you need business logic on this function
     * but for validation find a regex or develop a one
     */
-    if (typeof onChange === 'function') {
-      this.props.onChange(e)
+    if (typeof onAfterChange === 'function') {
+      onAfterChange(e)
     }
     /**
      * please catch value from this call back
      */
     if (typeof getValue === 'function') {
-      this.props.getValue(e.target.value)
+      getValue(e.target.value)
```

Comment:
why do we need a getValue function, can't it be obtained from onAfterChange

Reply:
get value will be same as of onAfterChange, but onAfterChange will be a good name for the function so we can use it instead of getValue